tests/test_project_admin/test_instructor_files/test_instructor_files.ts

```diff
@@ -237,14 +237,96 @@ describe('InstructorFiles.vue', () => {
         wrapper.findAll('.delete-file').at(0).trigger('click');
         await wrapper.vm.$nextTick();
 
-        expect(wrapper.find('.file-to-delete').text()).toContain(instructor_file_1.name);
+        expect(wrapper.find('.files-to-delete').findAll('li').length).toEqual(1);
+        expect(wrapper.find('.filename').text()).toContain(instructor_file_1.name);
 
         wrapper.find('.modal-delete-button').trigger('click');
         await wrapper.vm.$nextTick();
 
         expect(delete_stub.calledOnce).toBe(true);
     });
 
+    test('Delete single file while batched files are selected', async () => {
+        let delete_stub = sinon.stub(instructor_file_1, 'delete');
+        delete_stub.callsFake(
+            async () => InstructorFile.notify_instructor_file_deleted(instructor_file_1));
+
+        wrapper.findAll('.single-instructor-file-component input').at(0).trigger('click');
+        await wrapper.vm.$nextTick();
+        wrapper.findAll('.single-instructor-file-component input').at(1).trigger('click');
+        await wrapper.vm.$nextTick();
+
+
+        expect(delete_stub.callCount).toEqual(0);
+        wrapper.findAll('.delete-file').at(0).trigger('click');
+        await wrapper.vm.$nextTick();
+
+        expect(wrapper.find('.files-to-delete').findAll('li').length).toEqual(1);
+        expect(wrapper.find('.filename').text()).toContain(instructor_file_1.name);
+
+        wrapper.find('.modal-delete-button').trigger('click');
+        await wrapper.vm.$nextTick();
+
+        expect(delete_stub.calledOnce).toBe(true);
+    });
+
+    test('Delete single file using batch delete', async () => {
+        let delete_stub = sinon.stub(instructor_file_1, 'delete');
+        delete_stub.callsFake(
+            async () => InstructorFile.notify_instructor_file_deleted(instructor_file_1));
+
+        wrapper.findAll('.single-instructor-file-component input').at(0).trigger('click');
+        await wrapper.vm.$nextTick();
+
+        wrapper.findAll('.single-instructor-file-component input').at(0).trigger('click');
+        await wrapper.vm.$nextTick();
+
```

Comment:
For this particular test, I think clicking on the checkbox just once is fine.
Suggest adding a test where you select multiple files for deletion, unselect a subset of those, then trigger the batch delete and make sure that the right files are getting deleted

Reply:
1f21b41